src/Discord.Net.WebSocket/Entities/Invites/SocketInvite.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Model = Discord.API.InviteEvent;
+
+namespace Discord.WebSocket
+{
+    /// <summary>
+    ///     Represents a WebSocket-based invite.
+    /// </summary>
+    public class SocketInvite : SocketEntity<string>
+    {
+        /// <summary>
+        ///     Gets the channel in which this invite was created.
+        /// </summary>
+        public ISocketMessageChannel Channel { get; private set; }
+
+        /// <summary>
+        ///     Gets the channel ID in which this invite was created.
+        /// </summary>
+        public ulong ChannelId { get; private set; }
+
+        /// <summary>
+        ///     Gets the guild in which this invite was created.
+        /// </summary>
+        public IGuild Guild { get; private set; }
+
+        /// <summary>
+        ///     Gets the guild ID in which this invite was created.
+        /// </summary>
+        public ulong GuildId { get; private set; }
+
+        /// <summary>
+        ///     Gets the unique identifier for this invite.
+        /// </summary>
+        public string Code { get; private set; }
+
+        /// <summary>
+        ///     Gets the user who created this invite.
```

Comment:
I don't believe that we expose `Optional<T>` too often on the surface-level of the library. Should probably cache this? Regardless, where a value is not guaranteed to be a returned, the XML docs should say `Gets the X... if available.` or something like that.

Reply:
Yeah I'm not sure how else to replace that though. Caching would help, and I will implement that, but that doesn't guarantee that the invite will be cached before an `INVITE_DELETE` is sent. In my opinion keeping it wrapped in an optional is the safest route.